src/serializer/ResidualFunctions.js

```diff
@@ -225,7 +225,12 @@ export class ResidualFunctions {
     for (let [funcBody, instances] of functionEntries) {
       let functionInfo = this.residualFunctionInfos.get(funcBody);
       invariant(functionInfo);
-      this._referentialize(functionInfo.unbound, instances);
+      let additionalFunctionInstances = instances.filter((instance) => rewrittenAdditionalFunctions.has(instance.functionValue))
+      if (additionalFunctionInstances && additionalFunctionInstances.length) {
+        invariant(additionalFunctionInstances.length === instances.length, additionalFunctionInstances.length);
```

Comment:
I assume that all instances of the code for an additional function will map to the FunctionValue specified as an input to Prepack, this is somewhat dubious but I have an invariant for it.